modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java

```diff
@@ -372,6 +383,82 @@ private void configureIndexer() {
         checkSuccessfulResponse(response, "Could not configure the indexing service");
     }
 
+    /**
+     * Loads the sample buckets through the REST API.
+     */
+    private void loadSampleBuckets() {
+        if (sampleBuckets.isEmpty()) {
+            logger().debug("No sample buckets to load, skipping.");
+            return;
+        }
+        logger().debug("Loading sample buckets: {}", sampleBuckets);
+
+        List<String> content = sampleBuckets
+            .stream()
+            .map(SampleBucket::getName)
+            .collect(Collectors.toList());
+
+        try {
+            @Cleanup Response response = doHttpRequest(
+                MGMT_PORT,
+                "/sampleBuckets/install",
+                "POST",
+                RequestBody.create(MediaType.get("application/x-www-form-urlencoded"), MAPPER.writeValueAsString(content)),
+                true
+            );
+            checkSuccessfulResponse(response, "Could not create sample buckets " + sampleBuckets);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException("Could  not encode sample buckets into JSON!", e);
+        }
+
+        logger().debug("Loaded sample buckets, now waiting for each to be ready.");
+
+        // In addition to just loading the buckets (which is eventually consistent), we wait until the bucket
+        // has been created properly. Note that we do not create a primary index manually, since this is done
+        // by the sample bucket loading inside the cluster.
+        for (SampleBucket sampleBucket : sampleBuckets) {
+            logger().debug("Waiting for sample bucket {} to be ready", sampleBucket);
+
+            long start = System.nanoTime();
+
+            // This is doing the first sanity check that it is loaded
+            waitUntilBucketReady(sampleBucket.getName(), false);
+
+            // Now run a N1QL query (all sample buckets have a primary index) until the
+            // doc count is equal to the one that is expected.
+            // If the query service is enabled, make sure that we only proceed if the query engine also
+            // knows about the bucket in its metadata configuration.
+            Unreliables.retryUntilTrue(5, TimeUnit.MINUTES, () -> {
```

Comment:
5 minutes is potentially a long time to be waiting. Could we possibly log (quite visibly) at least before this step, and potentially at regular intervals if it's taking a long time?

Reply:
Yes, I'll add that. In my testing locally I had it to one minute but that timed out, so I figured going longer will make it more reliable.